### ISSUE TITLE
cannot run tad javascript error in main process (macosx)

### ISSUE THREAD

**Issue by abazabaaa** [antonycourtney/tad#167]
I am running macOS 11.5.2 and got the following error while trying to open a csv; I assume the issue is that duckdb is built of osx12 and that is the source of the error?

Uncaught Exception:
Error: dlopen(/Applications/Tad.app/Contents/Resources/app.asar.unpacked/node_modules/ac-node-duckdb/build/Release/node-duckdb-addon.node, 1): Symbol not found: __ZNKSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv
Referenced from: /Applications/Tad.app/Contents/Resources/app.asar.unpacked/node_modules/ac-node-duckdb/build/Release/./libduckdb.dylib (which was built for Mac OS X 12.0)
Expected in: /usr/lib/libc++.1.dylib

at process.func [as dlopen] (node:electron/js2c/asar_bundle:5:1812)
at Object.Module._extensions..node (node:internal/modules/cjs/loader:1199:18)
at Object.func [as .node] (node:electron/js2c/asar_bundle:5:2039)
at Module.load (node:internal/modules/cjs/loader:988:32)
at Module._load (node:internal/modules/cjs/loader:829:12)
at Function.c._load (node:electron/js2c/asar_bundle:5:13343)
at Module.require (node:internal/modules/cjs/loader:1012:19)
at require (node:internal/modules/cjs/helpers:102:18)
at Object.<anonymous> (/Applications/Tad.app/Contents/Resources/app.asar/node_modules/ac-node-duckdb/src/addon-bindings/connection-binding.ts:12:24)
at Module._compile (node:internal/modules/cjs/loader:1116:14)

**Comment by antonycourtney** [comment]
Yeah, this appears to be some incompatibility with C++ runtime libraries used by the DuckDb library built for Tad.
You're correct -- I built Tad on Mac OS X 12.3.1.  It's a bummer that libc++ doesn't seem to be backwards compatible between MacOS 12 and MacOS 11.
My only suggestion is to either upgrade to MacOS 12, or try building Tad from source on MacOS 11.
Sorry I can't offer a better answer!

**Comment by antonycourtney** [comment]
I am pleased to report that, thanks to setting up CI on Azure pipelines, I've managed to build an installer for Tad 0.10.0 that works on macOS 11 and macOS 12 (I've tested it on both). Since there is no change in the Tad source code, I've updated the DMG on github Releases without bumping the minor version.  Head on over to the [github Releases](https://github.com/antonycourtney/tad/releases/tag/v0.10.0) and download [Tad-0.10.0.dmg](https://github.com/antonycourtney/tad/releases/download/v0.10.0/tad-0.10.0.dmg) for a DMG that should work on Big Sur or Monterey!
